build/filter.go

```diff
@@ -724,6 +726,39 @@ func (in *DRBDLinkUtilizationFilter) Filter(system *v1.System, deployment *Deplo
 	return nil
 }
 
+type PlatformNetworkFilter interface {
+	Filter(platform_network *v1.PlatformNetwork, deployment *Deployment) error
+}
+
+// Filter for PlatformNetworks
+// This will filter out platformnetworks of type oam, mgmt and admin
+type CoreNetworkFilter struct {
+}
+
+func NewCoreNetworkFilter() *CoreNetworkFilter {
+	return &CoreNetworkFilter{}
+}
+
+func (in *CoreNetworkFilter) Filter(platform_network *v1.PlatformNetwork, deployment *Deployment) error {
+	if platform_network == nil {
```

Comment:
Isn't it always nil? What if it is not nil?

Reply:
If we want to introduce more filters, isn't it handled by newFilter.Filter()? Passing a new nil argument and set up a condition is quite weird. I understand it works because we have the "d.PlatformNetworks = nets" before, suggest:
1 Follow the existing filter's pattern: Filter(profile, deployment), assigning the networks to the deployment in/after the filter
or
2 remove the always nil argument and the condition